### ISSUE TITLE
Two Suggestions

### ISSUE THREAD

**Issue by Callmejp** [eth-sri/eran#4]
Hi!
l think Eran is great after l used it a lot.
But l have met some problems when l use it.

First, if my network has the Sigmoid activation in the last layer it will occur an error. Should the code in line 310 in deeppoly_nodes.py change to the code in line 260? And l find that the Sigmoid networks you offered have the Affine in the last layer. So maybe you already know this problem?

Second, why the MaxPool operation in read_net_file.py doesn't have its own stride_arg like the Conv2D operation? 

Thanks for reading!

**Comment by GgnDpSngh** [comment]
Hi,

Thanks for pointing this out. We will add support for both in the coming days and let you know.

Cheers,
Gagandeep Singh

**Comment by WenJieWan** [comment]
Hi!
i want to use my cifar model to verify the network, but i have met a problem when l use Eran.
my model file type is h5 ,so i use this convert method (https://github.com/Callmejp/DiplomaProject/tree/master/ChestNet) to extract weights by modifying the line 66 in tensorflow_translator.py to name. 
I don't know whether it successfully convert model.
Although it can run in the eran.analyze_box function, it can't run in the verify_network_with_milp function.  The gurobi optimizer doesn't has a feasible solution, because the status of model
is equal to GRB.Status.INFEASIBLE. It will cause a error "Unable to retrieve attribute objval".
So, can you tell me why my model can't compute by milp and how to modify it to compute successfully. 
My network structure is shown as follows：

model = Sequential()
model.add(Convolution2D(nb_filters, kernel_size, padding='same', activation='relu', input_shape=input_shape))
model.add(Convolution2D(nb_filters, kernel_size, padding='same', activation='relu'))
model.add(MaxPooling2D())
model.add(Convolution2D(nb_filters2, kernel_size, padding='same', activation='relu'))
model.add(Convolution2D(nb_filters2, kernel_size, padding='same', activation='relu'))
model.add(MaxPooling2D())
model.add(Flatten())
model.add(Dense(64, activation='relu'))
model.add(Dense(nb_classes, activation='sigmoid'))
model.compile(optimizer='Adam',loss='categorical_crossentropy',metrics=['accuracy'])

Thanks for reading!

**Comment by GgnDpSngh** [comment]
Hi,

We have added support for sigmoid and tanh activation in the last layer. We will add support for the other issue soon.

Cheers,

**Comment by Callmejp** [comment]
Great!
You have my appreciation！

---

Just out of curiosity, l want to ask a question and you don't have to answer it.
Will there be an accurate verification method that can be adapted to large-scale DNNs?
After all, it has been proved that it's a NPC problem.

**Comment by GgnDpSngh** [comment]
Hi,

We have added support for handling custom strides in the Maxpool. Regarding your question, we think that accurate verification can be designed for large scale DNNs by a combination of algorithmic and performance improvements that exploit the structure of the verification problem.

Cheers,
Gagandeep Singh
